### PR TITLE
add stacktrace component

### PR DESCRIPTION
**Pull request by oconn**
# PR for Issue #190  .

## PR Status

**READY**

## Changes proposed in the pull request:

- Addition of stack trace component

## Breaking changes?

- None

## Screenshot of feature:

image

@emilyseibert @oconn @dehli

### CODE REVIEW

File: less/panels/simulator.less [hunk 1]
```diff
@@ -28,12 +28,40 @@
...
+
+            .no-stack-trace-message {
+                font-family: monospace;
+                color: @white;
+                text-align: center;
+                font-size: 20px;
+            }
+
+            .stack-trace {
+                font-family: monospace;
```
Comment (emilyseibert) [inline less/panels/simulator.less]: can we the font-family set higher on the parent

File: src/cljs/wombats_web_client/components/tabbed_container.cljs [hunk 2]
```diff
@@ -5,22 +5,25 @@
 ;; Render Methods
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-(defn- render-tabs [{:keys [tabs index on-index-change]}]
-  (map-indexed (fn [tab-index {:keys [label]}] 
+(defn- render-tabs [{:keys [tabs index on-index-change] :as props}]
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/components/tabbed_container.cljs]: but you don't use `props` :(

File: src/cljs/wombats_web_client/components/tabbed_container.cljs [hunk 3]
```diff
@@ -5,22 +5,25 @@
...
-                                   :on-click #(on-index-change tab-index)} 
-                  label])
+                                   :on-click #(on-index-change tab-index)}
+                  label
+                  (when notifications
+                    [:span.notifications (notifications)])])
                tabs))
 
 (defn- render [{:keys [tabs index on-index-change] :as props}]
+  (prn tabs)
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/components/tabbed_container.cljs]: delete

File: src/cljs/wombats_web_client/panels/simulator.cljs [hunk 4]
```diff
@@ -24,19 +24,31 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 (defn- get-player [sim-state]
-  (let [players (:players @sim-state)
+  (let [players (:players (if (instance? reagent.ratom/Reaction sim-state)
```
Comment (emilyseibert) [inline src/cljs/wombats_web_client/panels/simulator.cljs]: why do we have to do `reagent.ratom` here?

File: less/panels/simulator.less (hunk 1 above)
Comment (oconn) [inline less/panels/simulator.less]: @emilyseibert its right here

Comment (emilyseibert) [inline less/panels/simulator.less]: *can we set the font-family higher on the parent
*set the font-family higher 
* yes i have word dyslexia

Comment (oconn) [inline less/panels/simulator.less]: @emilyseibert right here is where you grammar-ed bad

File: src/cljs/wombats_web_client/components/tabbed_container.cljs (hunk 2 above)
Comment (emilyseibert) [inline src/cljs/wombats_web_client/components/tabbed_container.cljs]: don't you but use `props`

File: src/cljs/wombats_web_client/components/tabbed_container.cljs (hunk 3 above)
Comment (emilyseibert) [inline src/cljs/wombats_web_client/components/tabbed_container.cljs]: dletee

File: src/cljs/wombats_web_client/panels/simulator.cljs (hunk 4 above)
Comment (emilyseibert) [inline src/cljs/wombats_web_client/panels/simulator.cljs]: `reagent.ratom` do have we why do here?

Comment (oconn) [inline src/cljs/wombats_web_client/panels/simulator.cljs]: This way you can access the state if you pass it an atom or state. Not a huge fan of this, will be refactored as a part of  #193

**Pull request merged**